### ISSUE TITLE
Rule proposal: no-object-methods-with-map

### ISSUE THREAD

**Issue by Lucas-C** [sindresorhus/eslint-plugin-unicorn#2024]
### Description

I recently got tricked by calling `Object.keys()` / `Object.values()` / `Object.entries()` on a `Map` object, and it took me a while to realize that it was wrong:
````javascript
const m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log('m.size:', m.size);
// Incorrect:
console.log('Object.keys(m):', Object.keys(m));
console.log('Object.values(m):', Object.values(m));
console.log('Object.entries(m):', Object.entries(m));
// Correct:
console.log('Array.from(m.keys()):', Array.from(m.keys()));
console.log('Array.from(m.values()):', Array.from(m.values()));
console.log('Array.from(m.entries()):', Array.from(m.entries()));
```

Would you be open to add a rule detecting the incorrect usage above?

### Fail

```js
var m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log(Object.keys(m));
```

```js
var m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log(Object.values(m));
```

```js
var m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log(Object.entries(m));
```


### Pass

```js
var m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log(Array.from(m.keys()));
```

```js
var m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log(Array.from(m.values()));
```

```js
var m = new Map();
m.set('a', 1);
m.set('b', 2);
console.log(Array.from(m.entries()));
```

### Additional Info

_No response_
